Update Sonar Scanner version for Improved Compatibility

SaikrishnaBairamoni opened the pull request:
<!-- Thanks for the contribution, this is awesome. -->

# PR Details
Updated Sonar Scanner to version 5.0.1.3006, compatible with Java 17.
## Description
This update addresses the need for keeping our environment aligned with the latest supported Java version for Sonar Scanner. The deprecated support for Java 11 necessitated this upgrade. These changes ensure our build and analysis processes remain efficient and up-to-date with current standards.
<!--- Describe your changes in detail -->

## Related GitHub Issue

<!--- This project only accepts pull requests related to open GitHub issues or Jira Keys -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please DO NOT name partially fixed issues, instead open an issue specific to this fix -->
<!--- Please link to the issue here: -->

## Related Jira Key

<!-- e.g. CAR-123 -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. --> Docker build process tested locally to ensure the successful installation of Java 17 and the new Sonar Scanner version to remove the deprecated warning 
- Ensured that existing functionalities are not affected by this upgrade.
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

JonSmet commented on Dockerfile:
```diff
@@ -255,8 +260,13 @@ RUN sed -i 's|http://archive.ubuntu.com|http://us.archive.ubuntu.com|g' /etc/apt
         sudo -u carma echo "source ~/.base-image/init-env.sh" >> /home/carma/.bashrc && \
         sudo -u carma echo "cd /opt/carma" >> /home/carma/.bashrc && \
+	apt-get update && \
+        apt-get install -y --no-install-recommends ${BASE_DEPS} && \
```
Are these two lines needed, since they have already been executed earlier in the Dockerfile?

JonSmet commented on Dockerfile:
```diff
@@ -108,13 +108,18 @@ ARG ROS_DEPS="apt-transport-https \
         python3-catkin-tools \
         python3-colcon-common-extensions \
         python3-future \
+        python3-lttng \
```
For some reason this PR looks like it is adding new packages to be installed in the Dockerfile under ROS_DEPS, but they are already included on the `develop` branch. Would you mind looking into resolving this, or alternatively starting clean and putting up a new PR?

JonSmet commented on Dockerfile:
```diff
@@ -108,13 +108,18 @@ ARG ROS_DEPS="apt-transport-https \
         libopenblas-dev \
         libpcap-dev \
         libpugixml-dev \
+        lttng-tools \
```
Not a specific comment on this code change, but please create a GitHub Issue in the carma-platform repository that we can link this PR to. It would be great if you can include the deprecated warning that we are receiving in that issue as well.

SaikrishnaBairamoni commented:
Closing this PR and opened new one due to branch issue (Addressed all above comments in new PR) : https://github.com/usdot-fhwa-stol/carma-base/pull/186
